new-cdc/eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java

```diff
@@ -108,22 +115,42 @@ public void start(Optional<BinlogFileOffset> binlogFileOffset, Consumer<M> event
     connectWithRetriesOnFail();
   }
 
-  private void handleWriteRowsEvent(Event event, Consumer<M> eventConsumer) {
+  private void handleWriteRowsEvent(Event event) {
     logger.debug("Got binlog event {}", event);
     offset = ((EventHeaderV4) event.getHeader()).getPosition();
     WriteRowsEventData eventData = event.getData();
     if (tableMapEventByTableId.containsKey(eventData.getTableId())) {
-      try {
-        eventConsumer.accept(writeRowsEventDataParser.parseEventData(eventData,
-                getCurrentBinlogFilename(), offset
-                )
-        );
-      } catch (IOException e) {
-        throw new RuntimeException("Event row parsing exception", e);
-      }
+
+      binlogEntryHandlers
+              .stream()
+              .filter(bh -> {
+                TableMapEventData tableMapEventData = tableMapEventByTableId.get(eventData.getTableId());
+
+                String database = tableMapEventData.getDatabase();
+                String table = tableMapEventData.getTable();
+
+                return checkSchemasAreEqual(bh, database) && bh.getSourceTableName().equalsIgnoreCase(table);
```

Comment:
change the return to  `return bh.isFor(database, table)`  - ie move logic into bh

Reply:
ok